control-base/src/main/java/fi/nls/oskari/control/data/LayerAndGroupOrderHandler.java

```diff
@@ -0,0 +1,172 @@
+        }
+        if(newLayerOrderMap == null) {
+        	newLayerOrderMap = new HashMap<>();
+        }
+    }
+
+    /**
+     * Handles updating the order and group of the given node i.e. layer or group.
+     * @param params
+     * @throws ActionException
+     */
+    @Override
+    public void handlePost(ActionParameters params) throws ActionException {
+        checkForAdminPermission(params);
+        log.debug("Updating layer/group order");
+        JSONObject orderJSON = getOrderJSON(params.getRequest());
+
+
+		try {
+			int parentID = orderJSON.getInt(KEY_PARENT);
+			JSONArray orders = orderJSON.getJSONArray(KEY_ORDERS);
+
+			// change main groups orders
+			if(parentID == -1){
+				for(int i=0;i<orders.length();i++){
+					JSONObject order = orders.getJSONObject(i);
+					int groupId = order.getInt(KEY_ID);
+					MaplayerGroup currentGroup = oskariMapLayerGroupService.find(groupId);
+					currentGroup.setOrderNumber(i);
+					oskariMapLayerGroupService.updateOrder(currentGroup);
+				}
+			} else {
+				for(int i=0;i<orders.length();i++) {
+					JSONObject order = orders.getJSONObject(i);
+					String type = order.getString(KEY_TYPE);
+					int id = order.getInt(KEY_ID);
+					if(TYPE_LAYER.equals(type)) {
+                        if(order.has(KEY_OLD_PARENT)) {
+                            oskariLayerService.updateGroup(id, order.getInt(KEY_OLD_PARENT), parentID);
+                        }
+					    oskariLayerService.updateOrder(id, parentID, i);
+					} else {
+						MaplayerGroup currentGroup = oskariMapLayerGroupService.find(id);
+						currentGroup.setOrderNumber(i);
+						oskariMapLayerGroupService.updateOrder(currentGroup);
+						if(order.has(KEY_OLD_PARENT)){
+						    oskariMapLayerGroupService.updateGroupParent(id,parentID);
+                        }
+					}
+				}
+			}
+
+
+
+        } catch (JSONException e) {
+        	log.warn(e);
+            throw new ActionException("Failed to read request!");
+        }
+
+        ResponseHelper.writeResponse(params, orderJSON);
+    }
+
+    
+    /**
+     * Read JSON from request
+     * @param req
+     * @return
+     * @throws ActionException
+     */
+    protected JSONObject getOrderJSON(HttpServletRequest req) throws ActionException {
+        try (InputStream in = req.getInputStream()) {
+            final byte[] json = IOHelper.readBytes(in);
+            final String jsonString = new String(json, StandardCharsets.UTF_8);
+            return new JSONObject(jsonString);
+        } catch (IOException e) {
+            log.warn(e);
+            throw new ActionException("Failed to read request!");
+        } catch (IllegalArgumentException | JSONException e) {
+            log.warn(e);
+            throw new ActionException("Invalid request!");
+        }
+    }
+    /**
+     * Commonly used with
+     * @param params
+     * @throws ActionException
+     */
+    private void checkForAdminPermission(ActionParameters params) throws ActionException {
```

Comment:
As noted previously, this functionality is available via `params.requireAdminUser();`